qiskit_experiments/calibration/experiments/spectroscopy.py

```diff
@@ -0,0 +1,350 @@
+            freq_guess = xdata[peak_idx]
+        if not sigma_guesses:
+            sigma_guesses = np.linspace(1e-6, abs(xdata[-1] - xdata[0]), 20)
+        if amplitude_bounds is None:
+            amplitude_bounds = [0.0, 1.1 * max(ydata)]
+        if sigma_bounds is None:
+            sigma_bounds = [0, abs(xdata[-1] - xdata[0])]
+        if freq_bounds is None:
+            freq_bounds = [0.9 * xdata[0], 1.1 * xdata[-1]]
+        if offset_bounds is None:
+            offset_bounds = [np.min(ydata), np.max(ydata)]
+
+        best_fit = None
+
+        lower = np.array([amplitude_bounds[0], sigma_bounds[0], freq_bounds[0], offset_bounds[0]])
+        upper = np.array([amplitude_bounds[1], sigma_bounds[1], freq_bounds[1], offset_bounds[1]])
+
+        # Perform fit
+        def fit_fun(x, a, sigma, x0_, b):
+            return a * np.exp(-((x - x0_) ** 2) / (2 * sigma ** 2)) + b
+
+        for sigma_guess in sigma_guesses:
+
+            fit_result = curve_fit(
+                fit_fun,
+                xdata,
+                ydata,
+                np.array([amp_guess, sigma_guess, freq_guess, offset_guess]),
+                sigmas,
+                (lower, upper),
+            )
+
+            if not best_fit:
+                best_fit = fit_result
+            else:
+                if fit_result["reduced_chisq"] < best_fit["reduced_chisq"]:
+                    best_fit = fit_result
+
+        best_fit["value"] = best_fit["popt"][2]
+        best_fit["stderr"] = (best_fit["popt_err"][2],)
+        best_fit["unit"] = experiment_data.data[0]["metadata"].get("unit", "Hz")
+        best_fit["label"] = "Spectroscopy"
+        best_fit["xdata"] = xdata
+        best_fit["ydata"] = ydata
+        best_fit["ydata_err"] = sigmas
+        best_fit["quality"] = self._fit_quality(
+            best_fit["popt"],
+            best_fit["popt_err"],
+            best_fit["reduced_chisq"],
+            xdata[0],
+            xdata[-1],
+        )
+
+        if plot:
+            ax = plot_curve_fit(fit_fun, best_fit, ax=ax)
+            ax = plot_scatter(xdata, ydata, ax=ax)
+            self._format_plot(ax, best_fit)
+            best_fit.plt = plt
+
+        return best_fit, None
+
+    @staticmethod
+    def _fit_quality(fit_out, fit_err, reduced_chisq, min_freq, max_freq) -> str:
+        """
+        Algorithmic criteria for whether the fit is good or bad.
+        A good fit has a small reduced chi-squared and the peak must be
+        within the scanned frequency range.
+
+        Args:
+            fit_out: Value of the fit.
+            fit_err: Errors on the fit value.
+            reduced_chisq: Reduced chi-squared of the fit.
+            min_freq: Minimum frequency in the spectroscopy.
+            max_freq: Maximum frequency in the spectroscopy.
+
+        Returns:
+            computer_bad or computer_good if the fit passes or fails, respectively.
+        """
+
+        if (
+            min_freq <= fit_out[2] <= max_freq
```

Comment:
It would be clearer to pass `fit_out[2]`, etc. to this function so they could have names like `fit_x0` and `fit_sigma` instead of just indices.

Reply:
Sounds good. See here: 44f7400